tests/components/slack/test_notify.py

```diff
@@ -1,7 +1,77 @@
 """Test slack notifications."""
-from unittest.mock import AsyncMock, Mock
-
-from homeassistant.components.slack.notify import SlackNotificationService
+import logging
+from unittest.mock import AsyncMock, Mock, patch
+
+from _pytest.logging import LogCaptureFixture
+import aiohttp
+from slack.errors import SlackApiError
+
+from homeassistant.components.slack.notify import (
+    CONF_DEFAULT_CHANNEL,
+    SlackNotificationService,
+    async_get_service,
+)
+from homeassistant.const import CONF_API_KEY, CONF_ICON, CONF_USERNAME
+from homeassistant.helpers.typing import HomeAssistantType
+
+MODULE_PATH = "homeassistant.components.slack.notify"
+
+
+async def test_get_service(hass: HomeAssistantType, caplog: LogCaptureFixture):
+    """Test async_get_service with exceptions."""
+    config = {
+        CONF_API_KEY: "12345",
+        CONF_DEFAULT_CHANNEL: "channel",
+    }
+
+    with patch(MODULE_PATH + ".aiohttp_client") as mock_session, patch(
+        MODULE_PATH + ".WebClient"
+    ) as mock_client, patch(
+        MODULE_PATH + ".SlackNotificationService"
+    ) as mock_slack_service:
+        mock_session.async_get_clientsession.return_value = session = Mock()
+        mock_client.return_value = client = AsyncMock()
+
+        # Normal setup
+        mock_slack_service.return_value = service = Mock()
+        assert await async_get_service(hass, config) == service
```

Comment:
Please start the tests with the core interfaces. We don't want to access any integration details in the tests.

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations

Reply:
It has been a while since I last contributed. So I missed that. #45561 should fix it.